### PR TITLE
54 text component

### PR DESCRIPTION
**Pull request by matthew-chirgwin**
## Status
**READY**

Closes: #54 

### Commit Message Title
* When opening this PR, the raiser should set the title of this PR to the first line of their desired commit message, e.g:
```
feat: add Text element component
```
* The reviewer should ensure that the first commit message field is of this form when performing the `squash and merge` from this page.

### Commit Message Description
```
- Adds Text element component, with all doc and tests
- Updates build and storybook to bring in carbon fonts and styling

Closes: #54

Signed-off-by: Matthew Chirgwin <chirmatt@uk.ibm.com>
```
* The reviewer should copy the above text into the extended description field when performing the `squash and merge` from this page.

## Checklist
- [X] Automated tests exist
- [X] Documentation exists [link]()
- [X] Local unit tests performed
- [X] Sufficient logging/trace
- [X] Desired commit message set as PR title and commit description set above

### CODE REVIEW

File: ui/src/Elements/Text/Text.view.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Text.defaultProps = {
+  type: BODY,
+  ...commonDefaultProps,
+};
+
+// high order component for other text types
+const withTypeStyle = (style, name) => {
+  // we want to set the type - so destructure and dont use type here
+  // eslint-disable-next-line no-unused-vars
+  const component = ({ type, ...others }) => <Text {...others} type={style} />;
```
Comment (nictownsend) [inline ui/src/Elements/Text/Text.view.js:65]: What's the type var used for?

Comment (matthew-chirgwin) [reply ui/src/Elements/Text/Text.view.js:65]: See comment on 63

File: ui/src/Elements/Text/Text.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.Text {
+    &--heading {
```
[the remark this answers is not in the document]
Comment (matthew-chirgwin) [reply ui/src/Elements/Text/Text.scss:2]: Fair point - shall update

File: ui/src/Elements/Text/Text.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.Text {
```
[the remark this answers is not in the document]
Comment (matthew-chirgwin) [reply ui/src/Elements/Text/Text.scss:1]: I have not, but will see what happens. Not a use case as was expecting

[the remark this answers is not in the document]
Comment (matthew-chirgwin) [reply ui/src/Elements/Text/Text.scss:1]: Answer - a hybrid of both!

Screenshot 2020-06-16 at 13 21 23

This is the a code, with a child of heading. Interesting (but logical) way of mixing styles! If we feel it is better to be explicit and only specify a string as child content I can do that as while interesting, I don't see a use case for this behaviour.

[the remark this answers is not in the document]
Comment (nictownsend) [reply ui/src/Elements/Text/Text.scss:1]: Nope let's leave as is, nice (expected) effect

File: ui/src/Elements/Text/Text.view.js (hunk 1 above)
Comment (nictownsend) [reply ui/src/Elements/Text/Text.view.js:65]: Oh yeah... <facepalm/>

**Pull request merged**